src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -82,9 +82,15 @@
     abstract public SamReader open(final File file);
 
     public SamReader open(final Path path) {
-        final SamInputResource r = SamInputResource.of(path, getPathWrapper());
+        return open(path, getPathWrapper(), Function.identity());
+    }
+
+    public SamReader open(final Path path,
+            Function<SeekableByteChannel, SeekableByteChannel> pathWrapper,
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper) {
+        final SamInputResource r = SamInputResource.of(path, pathWrapper);
         final Path indexMaybe = SamFiles.findIndex(path);
-        if (indexMaybe != null) r.index(indexMaybe);
+        if (indexMaybe != null) r.index(indexMaybe, indexWrapper);
         return open(r);
     }
```

Comment:
One way would be to copy the xor wrapper from GATK and check that we can read an XOR'd index.